### ISSUE TITLE
RevEng: Provide default context name when unspecified

### ISSUE THREAD

**Issue by bricelam** [aspnet/EntityFramework#3227]
The following command...
```ps1
Scaffold-DbContext 'Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=Northwind;Integrated Security=True' EntityFramework.SqlServer
```
...throws the following exception.
```
System.ArgumentException: The string argument 'dbContextClassName' cannot be empty.
   at Microsoft.Data.Entity.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.Data.Entity.Relational.Design.ReverseEngineering.ReverseEngineeringGenerator.CheckOutputFiles(String projectPath, String relativeOutputPath, String dbContextClassName, IModel metadataModel)
   at Microsoft.Data.Entity.Relational.Design.ReverseEngineering.ReverseEngineeringGenerator.GenerateAsync(ReverseEngineeringConfiguration configuration, CancellationToken cancellationToken)
   at Microsoft.Data.Entity.Design.DatabaseOperations.ReverseEngineerAsync(String provider, String connectionString, String outputDir, String dbContextClassName, Boolean useFluentApiOnly, CancellationToken cancellationToken)
   at Microsoft.Data.Entity.Design.OperationExecutor.<ReverseEngineerImpl>d__19.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.Data.Entity.Design.OperationExecutor.OperationBase.<>c__DisplayClass4_0`1.<Execute>b__0()
   at Microsoft.Data.Entity.Design.OperationExecutor.OperationBase.Execute(Action action)
The string argument 'dbContextClassName' cannot be empty.
```

**Comment by lajones** [comment]
@bricelam I'm not seeing it on my box. And it's weird - there are already a couple of layers of default - see https://github.com/aspnet/EntityFramework/blob/dev/src/EntityFramework.Relational.Design/ReverseEngineering/ReverseEngineeringGenerator.cs#L72 and https://github.com/aspnet/EntityFramework/blob/dev/src/EntityFramework.SqlServer.Design/ReverseEngineering/Configuration/SqlServerModelConfiguration.cs#L30

**Comment by bricelam** [comment]
@lajones Were you using the DNX or NuGet commands? PowerShell might be supplying an empty string instead of null.

**Comment by lajones** [comment]
AH, DNX. Let me look into that.

**Comment by lajones** [comment]
Checked in with commit 1e6c8c9. Please re-open if any problems.

**Comment by lajones** [comment]
Note: further fix for issue #3047

**Comment by bricelam** [comment]
Re-opening. Now I get the following.
```C#
public partial class  : DbContext
{
    // ...
}
```

**Comment by bricelam** [comment]
P.S. Code file is correct--`NorthwindContext.cs`.

**Comment by rowanmiller** [comment]
@lajones is this fixed now?

**Comment by lajones** [comment]
Signed off but not checked in. Per email am trying to include into beta-8 with table selection changes.

**Comment by lajones** [comment]
Fixed with PR #3356, commit 1c2916b.
